dev/PushNotifications/PushNotificationsLongRunningTask.StartupTask/winmain.cpp

```diff
@@ -0,0 +1,47 @@
+﻿#include <windows.h>
+#include "pch.h"
+
+#include "../PushNotifications-Constants.h"
+
+#include <NotificationsLongRunningProcess_h.h>
+#include <wil/resource.h>
+
+inline bool isRetriableRpcError(HRESULT hr)
+{
+    return (hr == HRESULT_FROM_WIN32(EPT_S_NOT_REGISTERED)) ||
+        (hr == HRESULT_FROM_WIN32(RPC_S_SERVER_UNAVAILABLE)) ||
+        (hr == HRESULT_FROM_WIN32(RPC_S_CALL_FAILED)) ||
+        (hr == HRESULT_FROM_WIN32(RPC_S_CALL_FAILED_DNE)) ||
+        (hr == HRESULT_FROM_WIN32(RPC_S_UNKNOWN_IF));
+}
+
+int WINAPI WinMain(HINSTANCE /*hInstance*/, HINSTANCE /*hPrevInstance*/, PSTR /*lpCmdLine*/, int /*nCmdShow*/)
+{
+    RETURN_IF_FAILED(::CoInitializeEx(nullptr, COINITBASE_MULTITHREADED));
+
+    auto scopeExit = wil::scope_exit(
+        [&]() { CoUninitialize(); });
+
+    unsigned int retries = 0;
+    bool isRpcRetriableError = true;
+    HRESULT hr = S_OK;
+
+    while (retries < 3 && isRpcRetriableError)
+    {
+        try
+        {
+            wil::com_ptr<INotificationsLongRunningPlatform> longRunningProcessPlatform
+                { wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+
+            isRpcRetriableError = false;
+        }
+        catch (...)
+        {
+            hr = wil::ResultFromCaughtException();
```

Comment:
Should this sleep or do some kind of back-off to let the system recover?

Reply:
My earlier comment seems to have been lost. :-(  When an OOP (outofproc) COM server comes to life, it has a certain amount of time to register class or activation factories (depending if classic COM or WinRT) and so there does not need to be a time allowance for the server to do its required registration(s) prior to doing a retry.

Reply:
Ok. So the guidance is to not have any kind of back-off for retrying?

Reply:
Sounds like it :) I think COM is ultimately doing the work here to give time for the process to come up and get it's components initialized. Atleast thats how I am interpreting the above.